providers/certificate.rb

```diff
@@ -0,0 +1,142 @@
+  EOH
+  
+  powershell_script @new_resource.name do
+    code codeScript
+    not_if notifScript
+  end
+end
+
+# acl_add is a modify-if-exists operation : not idempotent
+action :acl_add do
+  # We can do everything in a powershell script resource
+  hash = nil
+
+  if ::File.exists?(@new_resource.source)
+    # source is a file so get hash from that
+    file = win_friendly_path(@new_resource.source)
+    password = ", \"#{@new_resource.pfx_password}\"" if file.downcase.end_with?('pfx')
+    hash = "(New-Object System.Security.Cryptography.X509Certificates.X509Certificate2 \"#{file}\"#{password}).GetCertHashString()"
+  else
+    # make sure we have no spaces in the hash string
+    hash = "\"#{@new_resource.source.gsub(/\s/, '')}\""
+  end
+  
+  codeScript = <<-EOH
+    #{getAclScript(hash)}
+  EOH
+  onlyifScript = <<-EOH
+    $hash = #{hash}
+    Test-Path "cert:\\#{@location}\\#{@new_resource.store_name}\\$hash"
+  EOH
+  
+  powershell_script @new_resource.name do
+    code codeScript
+    only_if onlyifScript
+  end
+end
+
+action :delete do
+  # We can do everything in a powershell script resource
+  subject = @current_resource.source.sub(/\*/, '`*') # escape any * in the source
+  certCommand = "gci cert:\\#{@location}\\#{@new_resource.store_name} | where { $_.Subject -like '*#{subject}*' -or $_.Thumbprint -eq '#{subject}'}"
+  
+  codeScript = <<-EOH
+    $store = New-Object System.Security.Cryptography.X509Certificates.X509Store "#{@new_resource.store_name}", ([System.Security.Cryptography.X509Certificates.StoreLocation]::#{@location})
+    $store.Open([System.Security.Cryptography.X509Certificates.OpenFlags]::ReadWrite)
+    foreach ($c in #{certCommand})
+    {
+      $store.Remove($c)
+    }
+    $store.Close()
+  EOH
+  onlyifScript = <<-EOH
+    (#{certCommand}).Count -gt 0
+  EOH
+  
+  powershell_script @new_resource.name do
+    code codeScript
+    only_if onlyifScript
+  end
+end
+
+def load_current_resource
+  @current_resource = Chef::Resource::WindowsCertificate.new(@new_resource.name)
+  @current_resource.source(@new_resource.source)
+  @current_resource.pfx_password(@new_resource.pfx_password)
+  @current_resource.private_key_acl(@new_resource.private_key_acl)
+  @current_resource.store_name(@new_resource.store_name)
+  @current_resource.user_store(@new_resource.user_store)
+  @location = @current_resource.user_store ? 'CurrrentUser' : 'LocalMachine'
+end
+
+private
+def getAclScript(hash)
+  if (!@new_resource.private_key_acl.nil? && @new_resource.private_key_acl.length > 0)
+    raise "ACL can only be set on local machine certificates" if @new_resource.user_store
+
+    # this PS came from http://blogs.technet.com/b/operationsguy/archive/2010/11/29/provide-access-to-private-keys-commandline-vs-powershell.aspx
+    setAclScript = <<-EOH
+      $hash = #{hash}
+      $storeCert = gci "cert:\\#{@location}\\#{@new_resource.store_name}\\$hash"
```

Comment:
true, but ```hash``` is either a simple declaration or an expression (line 68) so for readability/sanity I declared $hash to force evaluation.